src/provider/provider.ts

```diff
@@ -0,0 +1,37 @@
+import keccak256 from 'keccak256';
+
+export type FourBytes = string;
+
+export default class EnsProvider {
```

Comment:
I don't see this class does a lot, so I would prefer to only have ens `Contract` class.

Reply:
The main reason is not to specify the url for etherium network each time I am creating a contract. Plus in the future, we might extend namicorn library functionality and this class can help us abstract some common code. Let's discuss this?